### PR TITLE
Dev Onboarding 2

### PR DESCRIPTION
**Pull request by dionyichia**
[//]: # "Feel free to customize this template to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Added supabase cilent and query script. Script pulls post and comments on render. Map script maps pullled data to component props.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

image

image

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @kylezryr

### CODE REVIEW

File: app/components/Comment.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const formattedDate = dateObject.toLocaleDateString('en-US', {
+    year: 'numeric',
+    month: 'long',
+    day: 'numeric',
+  });
+
+  return (
+    <div className="comment">
+      <div className="comment-head">
+        <UserProfile {...props.user} />
```
Comment (kylezryr) [inline app/components/Comment.tsx:21]: great way of handling passing props here i love it

File: supabase/query.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { supabase } from '@/supabase/client';
+import { Comment, Post } from './types';
+
+export class ApiService {
```
Comment (kylezryr) [inline supabase/query.ts:4]: perfect, this is exactly how we do supabase queries! havent seen people use a class before, usually we just define and export each query as a function but this works too!

File: app/utils/mapper.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      user: {
+        username: comment.username,
+      },
+      date: comment.date,
+      description: comment.comment,
+    })),
+  };
+}
+
+export async function mapPostsToPostProps(posts: Post[]): Promise<PostProps[]> {
```
Comment (kylezryr) [inline app/utils/mapper.ts:18]: great work here, super clean handling of processing the db data!

**Review by kylezryr — CHANGES_REQUESTED** [review]
amazing work dion! super clean code and very well written, no complaints here! my only suggestion would be to try and convert some (or all if u have time) of your components to use StyledComponents instead of plain CSS to get some practice with your team's styling convention! u can make the changes to this branch and re request a review from me when done
